app/src/main/java/com/rockthevote/grommet/ui/MainActivity.java

```diff
@@ -74,6 +86,50 @@ protected void onCreate(Bundle savedInstanceState) {
         setSupportActionBar(toolbar);
         requestGPSPermission();
         hockeyAppHelper.checkForUpdates(this);
+
+        viewModel = new ViewModelProvider(
+                this,
+                new MainActivityViewModelFactory(rockyService)
+        ).get(MainActivityViewModel.class);
+
+        observeState();
+    }
+
+    private void observeState() {
+        viewModel.getState().observe(this, mainActivityState -> {
+
+            // Reset base states
+            upploadButton.setEnabled(true);
+            failedRegistrationsContainer.setVisibility(View.GONE);
+            failedRegistrations.setText("");
+
+            if (mainActivityState instanceof MainActivityState.Content) {
+                MainActivityState.Content content = (MainActivityState.Content) mainActivityState;
+
+                String pendingUploads = String.format(
+                        Locale.getDefault(),
```

Comment:
Not sure if this is just needed in order to format date, but we don't have to worry about language on this page, only within the registration process. This is always english.

Reply:
Eh, it was a lint warning so 🤷‍♂️, not needed, but not harmful

Reply:
works for me